lib/firenest/topology.ex

```diff
@@ -215,57 +173,57 @@ defmodule Firenest.Topology do
   """
   @spec disconnect(t, node) :: true | false | :ignored
-  def disconnect(topology, node) do
+  def disconnect(topology, node) when is_atom(topology) and is_atom(node) do
     adapter!(topology).disconnect(topology, node)
   end
 
   @doc """
-  Monitors the given `name` in `node` through `topology`.
+  Syncs the given `pid` across the topology using its name.
 
-  Once the monitored process crashes (or the connection to
-  the node the monitored process runs on is lost), a message
-  with the format described below is delivered to the caller
-  process:
+  This function is the building block for building static services
+  on top of the topology. It allows the current process to know whenever
+  another process with the same name goes up or down in the topology
+  as long as processes call `sync_named/2`.
 
-      {:DOWN, ref, :process, {name, node}, reason}
+  This function returns `{:ok, nodes}` in case the given pid has not
+  been synced yet, `{:error, {:already_synced, pid}}` otherwise.
+  `nodes` is a list of tuples with the first element with the node
+  name as an atom and the second element is a term used to version
+  that node name. Only the nodes that are known to have a service
+  with the same `name` running and that have already called `sync_named/2`
+  will be included in the list.
 
-  Where `ref` is the reference returned by `monitor/3` and `reason`
-  is the reason the process exited.
+  Once this function is called, the given process `pid` will receive
+  two messages with the following guarantees:
 
-  This operations always return a `reference`, regardless if
-  there is a connection to `node` or not. In case there is no
-  connection, a DOWN message will be delivered to the caller with
-  reason of `:noconnection`. In case there is no process with `name`
-  in `node`, a DOWN message with reason of `:noproc` is sent.
-  """
-  @spec monitor(t, node, name) :: reference
-  def monitor(topology, node, name) do
-    adapter!(topology).monitor(topology, node, name)
-  end
+    * `{:named_up, node, id, name}` is delivered whenever a process
+      with name `name` is up on the given `node-id` pair. The message
+      is guaranteed to be delivered after the node is added to the list
+      returned by `nodes/2`.
 
-  @doc """
-  Subscribes `pid` to the `topology` `:nodeup` and `:nodedown` events.
+  * `{:named_down, node, id, name}` is delivered whenever a process
+      with name `name` is down on the given `node-id` pair. It can be
+      deivered when such processes crashes or when there is a disconnection.
+      The message is guaranteed to be delivered after the node is removed
+      from the list returned by `nodes/2`. Note the topology may not
+      necessarily guarantee that no messages are received from `name`
+      after this message is sent.
 
-  See the module documentation for a description of events and their
-  guarantees.
+  In a case node loses connection and reconnects (either due to network
```

Comment:
Just a grammar thing here, I think you meant "In the case where a node loses connection and reconnects..." or perhaps more simply "If the connection to a node is lost, perhaps due to a network partition or crash, and then reestablished..."